File: src/test/java/com/n26/service/StatisticsComputationRoundedUpScaling.java
package com.n26.service;

import com.n26.model.Statistics;
import org.junit.Before;
import org.junit.Test;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Arrays;

import static org.junit.Assert.assertEquals;

public class StatisticsComputationRoundedUpScaling {

    private StatisticsComputation statisticsComputation;
    private ArrayList<BigDecimal> amounts;


    @Before
    public void setData() {
        amounts = new ArrayList<>(Arrays.asList(BigDecimal.valueOf(5.0), BigDecimal.valueOf(4.0), BigDecimal.valueOf(2.0)));
        statisticsComputation = new StatisticsComputation();
    }

    @Test
    public void computeSumRoundUp(){
        Statistics statistics = statisticsComputation.computeStatistics(amounts);
        assertEquals(BigDecimal.valueOf(1100,2),statistics.getSum());
    }

    @Test
    public void computeAverageRoundUp(){
        Statistics statistics = statisticsComputation.computeStatistics(amounts);
        assertEquals(BigDecimal.valueOf(367,2),statistics.getAvg());
    }
}

File: src/test/kotlin/com/service/StatisticsComputationRoundedUpScaling.kt
package com.service

import java.math.BigDecimal
import com.service.serviceImpl.StatisticsComputationImpl
import org.junit.Assert
import org.junit.Test

class StatisticsComputationRoundedUpScaling {
    private var statisticsComputation: StatisticsComputation = StatisticsComputationImpl()
    private var amounts: List<BigDecimal> = listOf(
        BigDecimal.valueOf(5.0),
        BigDecimal.valueOf(4.0),
        BigDecimal.valueOf(2.0))

    @Test
    fun computeSumRoundUp() {
        val sum = statisticsComputation.computeStatistics(amounts).sum
        Assert.assertEquals(BigDecimal.valueOf(1100, 2), sum)
    }

    @Test
    fun computeAverageRoundUp() {
        val avg = statisticsComputation.computeStatistics(amounts).avg
        Assert.assertEquals(BigDecimal.valueOf(367, 2), avg)
    }
}